test/validation/api/random/random.c

```diff
@@ -18,22 +20,48 @@ static void random_test_get_size(void)
 	 * arbitrarily (although considered sane) and neither platforms or
 	 * applications are not required to use it.
 	 */
-	int32_t ret, timeout_ns = 1 * ODP_TIME_MSEC_IN_NS, sleep_ns = 100;
-	uint32_t bytes = 0;
+	int32_t ret, timeout_ns, sleep_ns;
+	uint32_t bytes;
 	uint8_t buf[32];
-
-	do {
-		ret = odp_random_data(buf + bytes, sizeof(buf) - bytes,
-				      ODP_RANDOM_BASIC);
-		bytes += ret;
-		if (ret < 0 || bytes >= sizeof(buf))
+	odp_random_kind_t kind;
+	int max_kind = odp_random_max_kind();
```

Comment:
Instead of testing all random kinds inside the same function, random kind could be made a function argument like in the new tests Jere added:
```
	ODP_TEST_INFO_CONDITIONAL(random_test_align_and_overflow_basic, check_kind_basic),
	ODP_TEST_INFO_CONDITIONAL(random_test_align_and_overflow_crypto, check_kind_crypto),
	ODP_TEST_INFO_CONDITIONAL(random_test_align_and_overflow_true, check_kind_true),
```

This way it's easier to notice skipped/failed tests.